repair/repair.cc

```diff
@@ -1233,14 +1233,6 @@ int repair_service::do_repair_start(sstring keyspace, std::unordered_map<sstring
             cfs = std::move(cfs), ranges = std::move(ranges), options = std::move(options), ignore_nodes = std::move(ignore_nodes)] () mutable {
         auto participants = get_hosts_participating_in_repair(db.local(), keyspace, ranges, options.data_centers, options.hosts, ignore_nodes).get();
 
-        if (db.local().get_config().wait_for_hint_replay_before_repair()) {
```

Comment:
If we are doing this, we might as well remove the `wait_for_hint_replay_before_repair` configuration option (introduced in 86d831b319c487dbf6da5f3f04935b397b018de0) because it becomes useless and will be removed anyway when we switch to the HTTP API. AFAIK, the commit 86d831b319c487dbf6da5f3f04935b397b018de0 is not a part of any release yet.

Reply:
Yes, let's do that before it is released. @piodul can you take it from here and do the proper revert. It won't be a clean revert now.

Reply:
OK, I'll prepare a proper revert.